src/routes/render/renderAuthenticatedImage.ts

```diff
@@ -0,0 +1,111 @@
+import { FastifyInstance } from 'fastify'
+import { Obj } from '../../types/types'
+import { normalizeContentType, transformPostgrestError } from '../../utils'
+import { Imgproxy } from '../../renderer/imgproxy'
+import { AxiosError } from 'axios'
+
+const { region, globalS3Bucket, globalS3Endpoint, storageBackendType, imgProxyURL } = getConfig()
+
+let storageBackend: GenericStorageBackend
+
+if (storageBackendType === 'file') {
+  storageBackend = new FileBackend()
+} else {
+  storageBackend = new S3Backend(region, globalS3Endpoint)
+}
+
+const imageRenderer = new Imgproxy(storageBackend, {
+  url: imgProxyURL || '',
+})
+
+const renderAuthenticatedImageParamsSchema = {
+  type: 'object',
+  properties: {
+    bucketName: { type: 'string', examples: ['avatars'] },
+    '*': { type: 'string', examples: ['folder/cat.png'] },
+  },
+  required: ['bucketName', '*'],
+} as const
+
+const renderImageQuerySchema = {
+  type: 'object',
+  properties: {
+    height: { type: 'number', examples: [100] },
+    width: { type: 'number', examples: [100] },
+    resize: { type: 'string', enum: ['fill', 'fit', 'fill-down', 'force', 'auto'] },
+  },
+} as const
+
+interface renderImageRequestInterface {
+  Params: FromSchema<typeof renderAuthenticatedImageParamsSchema>
+  Querystring: FromSchema<typeof renderImageQuerySchema>
+}
+
+export default async function routes(fastify: FastifyInstance) {
+  const summary = 'Render an authenticated image with the given transformations'
+  fastify.get<renderImageRequestInterface>(
+    '/authenticated/:bucketName/*',
+    {
+      schema: {
+        params: renderAuthenticatedImageParamsSchema,
+        summary,
+        response: { '4xx': { $ref: 'errorSchema#', description: 'Error response' } },
+        tags: ['object'],
+      },
+    },
+    async (request, response) => {
+      const { bucketName } = request.params
+      const objectName = request.params['*']
+
+      const objectResponse = await request.postgrest
+        .from<Obj>('objects')
+        .select('id')
+        .match({
+          name: objectName,
+          bucket_id: bucketName,
+        })
+        .single()
+
+      if (objectResponse.error) {
+        const { status, error } = objectResponse
+        request.log.error({ error }, 'error object')
+        return response.status(400).send(transformPostgrestError(error, status))
+      }
+
+      const s3Key = `${request.tenantId}/${bucketName}/${objectName}`
+
+      try {
+        const imageResponse = await imageRenderer.transform(globalS3Bucket, s3Key, request.query)
+
+        response
+          .status(imageResponse.status)
+          .header('Accept-Ranges', 'bytes')
+          .header('Content-Type', normalizeContentType(imageResponse.headers['content-type']))
+          .header('Cache-Control', imageResponse.headers['cache-control'])
+          .header('Content-Length', imageResponse.headers['content-length'])
+          .header('ETag', imageResponse.headers['etag'])
```

Comment:
Send the last-modified header too like [here](https://github.com/supabase/storage-api/blob/a14d627002c0fbc4bb408757a01f68a4880f0eb3/src/routes/object/getObject.ts#L95)?

Reply:
we don't have this info as we don't store it currently. 
It could be a good thing to store the value in the metadata in another PR